Modules/selectmodule.c

```diff
@@ -1526,7 +1521,19 @@ select_epoll_poll_impl(pyEpoll_Object *self, PyObject *timeout_obj,
             return NULL;
         }
 
-        deadline = _PyTime_GetMonotonicClock() + timeout;
+        if (timeout >= 0) {
+            deadline = _PyTime_GetMonotonicClock() + timeout;
+        }
+    }
+
+    /* Kernel doesn't make any difference for negative values, but -1 is
+       documented as a way to block indefinitely in the epoll_wait()
+       documentation, so set ms to -1 if the value is negative.
+
+       Note that we didn't use INFTIM here since it's non-standard and
+       doesn't available under Linux. */
+    if (ms < 0) {
```

Comment:
It is better to move this check above, closer to other check for `ms`.

Reply:
Moved and tweaked the comment a bit.